extras/plot_throughput/plot_data.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+import pandas as pd
+import glob
+import subprocess
+import os
+
+# Create some useful directories
+try:
+    os.mkdir('clean')
+except OSError as error:
+    print(error)
+try:
+    os.mkdir('plots')
+except OSError as error:
+    print(error)
+
+# Get all csv input filenames
+filenames = glob.glob('*.csv')
+print("filenames = {}".format(" ".join(filenames)) )
+
+output_peaks = 'clean/peaks.csv'
+
+peaks = pd.DataFrame()
+
+for input_csv in filenames :
+    # Ignore the peaks file
+    if input_csv == output_peaks :
+        continue
+
+    # Create output filename in clean dir
+    output_csv = 'clean/{}'.format(input_csv)
+
+    # Read raw input data
+    raw = pd.read_csv(input_csv)
+    print("\nRead ", len(raw.index), " datapoints from ", input_csv)
+
+    # Get int value from input directory name
+    raw['mu'] = raw['directory'].str.extract('(\d+)').astype(int)
+
+    # Calculate throughput
+    raw['throughput (events/s)'] = raw.apply(lambda x: x['processed_events'] * 1e9 / x['processing_time'] , axis=1)
+    raw = raw[['mu', 'threads', 'throughput (events/s)']]
+
+    # Calculate mean throughput and respective standard deviation for each mu/thread pair
+    clean = raw.groupby(['mu', 'threads']).mean().fillna(0.0)
+    clean['stddev (events/s)'] = raw.groupby(['mu', 'threads']).std().fillna(0.0)['throughput (events/s)']
+
+    clean = clean.reset_index('mu')
+
+    # Separate each mu value to different columns
+    groups = clean.groupby('mu')
+
+    final = pd.DataFrame()
+    for x in groups.groups:
+        final['throughput mu{} (events/s)'.format(x)] = groups.get_group(x)['throughput (events/s)']
+        final['stddev mu{} (events/s)'.format(x)] = groups.get_group(x)['stddev (events/s)']

```

Comment:
This can be written more elegantly, and I do not blame you for not spotting this:

```suggestion
    final = clean.unstack(level=0)
    final.columns = [f"{x}_{y}" for _, x, y in final.columns.to_flat_index()]
    final.to_csv(output_csv)
```

Reply:
Resolving this conversation as the gnuplot script is a goner so I no longer need this column organization part.